be/src/formats/parquet/encoding_dict.h

```diff
@@ -296,37 +282,34 @@ class DictDecoder<Slice> final : public Decoder {
     Status skip(size_t values_to_skip) override {
         //TODO(Smith) still heavy work load
         _indexes.reserve(values_to_skip);
-        _index_batch_decoder.GetBatch(&_indexes[0], values_to_skip);
+        _rle_batch_reader.GetBatch(&_indexes[0], values_to_skip);
         return Status::OK();
     }
 
     Status next_batch(size_t count, ColumnContentType content_type, Column* dst) override {
-        _indexes.reserve(count);
-        _index_batch_decoder.GetBatch(&_indexes[0], count);
-
         switch (content_type) {
         case DICT_CODE: {
-            auto ret = dst->append_numbers(&_indexes[0], count * SIZE_OF_DICT_CODE_TYPE);
-            if (UNLIKELY(!ret)) {
-                return Status::InternalError("DictDecoder append numbers to column failed");
+            FixedLengthColumn<int32_t>* data_column;
```

Comment:
suggest to use RunTimeColumnType<TYPE_INT> instead of FixedLengthColumn<int32_t>

```cpp
using DictCodeColumn = RunTimeColumnType<TYPE_INT>;
```